numpy/core/src/multiarray/dtype_transfer.c

```diff
@@ -397,6 +397,10 @@ _strided_to_strided_contig_align_wrap(char *dst, npy_intp dst_stride,
             tobuffer(bufferin, inner_src_itemsize, src, src_stride,
                                     NPY_LOWLEVEL_BUFFER_BLOCKSIZE,
                                     src_itemsize, todata);
+            if (init_dest) {
+                memset(bufferout, 0,
+                        dst_itemsize*NPY_LOWLEVEL_BUFFER_BLOCKSIZE);
+            }
```

Comment:
I suppose the idea of this was to avoid paying the cost of the branch during casting. However, it only runs for every `NPY_LOWLEVEL_BUFFER_BLOCKSIZE == 128` elements, so that's probably not a big deal.

Reply:
Yeah, and it's easily predicted too (if the compiler doesn't take care of it already).